File: backend/_archive/kb-document/src/main/java/com/knowledge/base/document/service/CategoryService.java
package com.knowledge.base.document.service;

import com.knowledge.base.document.dto.CategoryDTO;
import com.knowledge.base.document.vo.CategoryVO;

import java.util.List;

/**
 * 分类Service接口
 *
 * <p>提供文档分类相关业务逻辑</p>
 *
 * @author 苏三
 * @since 1.0.0
 */
public interface CategoryService {

    /**
     * 创建分类
     *
     * @param categoryDTO 分类信息
     * @return 分类ID
     */
    Long createCategory(CategoryDTO categoryDTO);

    /**
     * 更新分类
     *
     * @param categoryDTO 分类信息
     * @return 是否成功
     */
    Boolean updateCategory(CategoryDTO categoryDTO);

    /**
     * 删除分类
     *
     * @param categoryId 分类ID
     * @return 是否成功
     */
    Boolean deleteCategory(Long categoryId);

    /**
     * 根据ID查询分类
     *
     * @param categoryId 分类ID
     * @return 分类信息
     */
    CategoryVO getCategoryById(Long categoryId);

    /**
     * 获取分类树
     *
     * @return 分类树
     */
    List<CategoryVO> getCategoryTree();

    /**
     * 获取子分类
     *
     * @param parentId 父分类ID
     * @return 子分类列表
     */
    List<CategoryVO> getChildren(Long parentId);

    /**
     * 移动分类
     *
     * @param categoryId   分类ID
     * @param newParentId 新父分类ID
     * @return 是否成功
     */
    Boolean moveCategory(Long categoryId, Long newParentId);

    /**
     * 获取所有分类
     *
     * @return 分类列表
     */
    List<CategoryVO> getAllCategories();
}

File: frontend/src/services/category.service.ts
import { http } from './request';
import { DocumentCategory, CategoryTree, PageParams } from '@/types';

export interface CategoryMoveParams {
  categoryId: string;
  targetParentId?: string;
  position?: number;
}

export const categoryService = {
  // 获取分类列表（平铺）
  getCategories: () => {
    return http.get<DocumentCategory[]>('/document/categories/list');
  },

  // 获取分类树
  getCategoryTree: () => {
    return http.get<CategoryTree[]>('/document/categories/tree');
  },

  // 获取一级分类（rootOnly，用于侧边栏）
  getRootCategories: () => {
    return http.get<CategoryTree[]>('/document/categories/children/0');
  },

  // 获取分类详情
  getCategory: (id: string) => {
    return http.get<DocumentCategory>(`/document/categories/${id}`);
  },

  // 创建分类
  createCategory: (data: {
    name: string;
    description?: string;
    icon?: string;
    parentId?: string;
    sort?: number;
  }) => {
    return http.post<DocumentCategory>('/document/categories', data);
  },

  // 更新分类（后端 PUT /categories，id 放 body）
  updateCategory: (id: string, data: Partial<DocumentCategory>) => {
    return http.put<DocumentCategory>('/document/categories', { ...data, id });
  },

  // 删除分类
  deleteCategory: (id: string) => {
    return http.delete(`/document/categories/${id}`);
  },

  // 移动分类（后端 PUT /categories/{id}/move?newParentId=）
  moveCategory: (params: CategoryMoveParams) => {
    const parentId = params.targetParentId ?? '0';
    return http.put(`/document/categories/${params.categoryId}/move`, null, {
      params: { newParentId: parentId },
    });
  },

  // 批量删除分类
  batchDeleteCategories: (ids: string[]) => {
    return http.delete('/document/categories/batch', { data: { ids } });
  },

  // 获取分类下的文档（暂无文档分页 + categoryId，避免无后端接口）
  getCategoryDocuments: (categoryId: string, params?: PageParams) => {
    return http.get('/document/documents/page', {
      params: { ...params, categoryId },
    });
  },

  // 获取分类统计
  getCategoryStats: () => {
    return http.get<Array<{
      categoryId: string;
      categoryName: string;
      documentCount: number;
      viewCount: number;
    }>>('/document/categories/stats');
  },

  // 搜索分类
  searchCategories: (keyword: string) => {
    return http.get<DocumentCategory[]>('/document/categories/search', { params: { keyword } });
  },
};

export default categoryService;
